internal/rukpak/convert/registryv1.go

```diff
@@ -53,92 +53,73 @@ func RegistryV1ToPlain(rv1 fs.FS, installNamespace string, watchNamespaces []str
 	}
 	reg.PackageName = annotationsFile.Annotations.PackageName
 
-	var objects []*unstructured.Unstructured
 	const manifestsDir = "manifests"
-
-	entries, err := fs.ReadDir(rv1, manifestsDir)
-	if err != nil {
-		return nil, err
-	}
-	for _, e := range entries {
+	if err := fs.WalkDir(rv1, manifestsDir, func(path string, e fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
 		if e.IsDir() {
-			return nil, fmt.Errorf("subdirectories are not allowed within the %q directory of the bundle image filesystem: found %q", manifestsDir, filepath.Join(manifestsDir, e.Name()))
+			if path == manifestsDir {
+				return nil
+			}
+			return fmt.Errorf("subdirectories are not allowed within the %q directory of the bundle image filesystem: found %q", manifestsDir, path)
 		}
-		fileData, err := fs.ReadFile(rv1, filepath.Join(manifestsDir, e.Name()))
+		manifestFile, err := rv1.Open(path)
 		if err != nil {
-			return nil, err
+			return err
 		}
-
-		dec := apimachyaml.NewYAMLOrJSONDecoder(bytes.NewReader(fileData), 1024)
-		for {
-			obj := unstructured.Unstructured{}
-			err := dec.Decode(&obj)
-			if errors.Is(err, io.EOF) {
-				break
-			}
-			if err != nil {
-				return nil, fmt.Errorf("read %q: %v", e.Name(), err)
+		defer func() {
+			if err := manifestFile.Close(); err != nil {
+				l.Error(err, "error closing file", "path", path)
```

Comment:
This _appears_ to be the only place `l` is used. Should `l := log.FromContext(ctx)` be moved down here for locality?

Reply:
(or even into the WalkDir function?

Reply:
If we did that, we'd have to repeatedly ask the context for the same logger over and over. It feels better (to me at least) to get the "logger to be used in the scope of this function" once, right at the very beginning of the function.